contracts/VeJoeStaking.sol

```diff
@@ -0,0 +1,287 @@
+            "VeJoeStaking: expected new _maxCap to be greater than 0 and leq to 100000"
+        );
+        maxCap = _maxCap;
+        emit UpdateMaxCap(msg.sender, _maxCap);
+    }
+
+    /// @notice Set baseGenerationRate
+    /// @param _baseGenerationRate the new baseGenerationRate
+    function setBaseGenerationRate(uint256 _baseGenerationRate) external onlyOwner {
+        require(
+            _baseGenerationRate < boostedGenerationRate,
+            "VeJoeStaking: expected new _baseGenerationRate to be less than boostedGenerationRate"
+        );
+        baseGenerationRate = _baseGenerationRate;
+        emit UpdateBaseGenerationRate(msg.sender, _baseGenerationRate);
+    }
+
+    /// @notice Set boostedGenerationRate
+    /// @param _boostedGenerationRate the new boostedGenerationRate
+    function setBoostedGenerationRate(uint256 _boostedGenerationRate) external onlyOwner {
+        require(
+            _boostedGenerationRate > baseGenerationRate,
+            "VeJoeStaking: expected new _boostedGenerationRate to be greater than baseGenerationRate"
+        );
+        boostedGenerationRate = _boostedGenerationRate;
+        emit UpdateBoostedGenerationRate(msg.sender, _boostedGenerationRate);
+    }
+
+    /// @notice Set boostedThreshold
+    /// @param _boostedThreshold the new boostedThreshold
+    function setBoostedThreshold(uint256 _boostedThreshold) external onlyOwner {
+        require(
+            _boostedThreshold <= 100,
+            "VeJoeStaking: expected new _boostedThreshold to be less than or equal to 100"
+        );
+        boostedThreshold = _boostedThreshold;
+        emit UpdateBoostedThreshold(msg.sender, _boostedThreshold);
+    }
+
+    /// @notice Set boostedDuration
+    /// @param _boostedDuration the new boostedDuration
+    function setBoostedDuration(uint256 _boostedDuration) external onlyOwner {
+        boostedDuration = _boostedDuration;
+        emit UpdateBoostedDuration(msg.sender, _boostedDuration);
+    }
+
+    /// @notice Deposits JOE to start staking for veJOE. Note that any pending veJOE
+    /// will also be claimed in the process.
+    /// @param _amount the amount of JOE to deposit
+    function deposit(uint256 _amount) external {
+        require(_amount > 0, "VeJoeStaking: expected deposit amount to be greater than zero");
+
+        if (getUserHasNonZeroBalance(msg.sender)) {
+            // If user already has staked JOE, we first send them any pending veJOE
+            _claim();
+
+            uint256 userStakedJoe = userInfos[msg.sender].balance;
+
+            userInfos[msg.sender].balance += _amount;
+
+            // User is eligible for boosted benefits if and only if all of the following are true:
+            // - User is not already currently receiving boosted benefits
+            // - `_amount` is at least `boostedThreshold / 100 * userStakedJoe`
+            if (userInfos[msg.sender].boostEndTimestamp == 0 && _amount * 100 >= boostedThreshold * userStakedJoe) {
```

Comment:
When will `userInfos[msg.sender].boostEndTimestamp > 0` **and** user has non zero balance? Seems like this line will never be hit